### PR TITLE
 Vpnaas: List IPSec Policies

### PR DESCRIPTION
**Pull request by simonre**
For #723 

Links to the line numbers/files in the OpenStack source code that support the
code in this PR:

https://github.com/openstack/neutron-vpnaas/blob/058469e1b99b647537a5228c6a384d93df5484df/neutron_vpnaas/db/vpn/vpn_db.py#L454

### CODE REVIEW

File: openstack/networking/v2/extensions/vpnaas/ipsecpolicies/requests.go [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+// List request.
+type ListOptsBuilder interface {
+	ToPolicyListQuery() (string, error)
+}
+
+// ListOpts allows the filtering of paginated collections through
+// the API. Filtering is achieved by passing in struct field values that map to
+// the IPSec policy attributes you want to see returned.
+type ListOpts struct {
+	ID                  string   `q:"id"`
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/vpnaas/ipsecpolicies/requests.go]: Same comment here as with the Services ListOpts.

File: openstack/networking/v2/extensions/vpnaas/ipsecpolicies/requests.go [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+	TenantID            string   `q:"tenant_id"`
+	Name                string   `q:"name"`
+	Description         string   `q:"description"`
+	ProjectID           string   `q:"project_id"`
+	AuthAlgorithm       string   `q:"auth_algorithm"`
+	EncapsulationMode   string   `q:"encapsulation_mode"`
+	EncryptionAlgorithm string   `q:"encryption_algorithm"`
+	PFS                 string   `q:"pfs"`
+	Lifetime            Lifetime `q:"lifetime"`
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/vpnaas/ipsecpolicies/requests.go]: Is this actually possible to query with?

File: openstack/networking/v2/extensions/vpnaas/ipsecpolicies/testing/requests_test.go [hunk 3]
```diff
@@ -0,0 +1,171 @@
...
+{
+    "ipsecpolicies": [
+		{
+			"name": "ipsecpolicy1",
+			"transform_protocol": "esp",
+			"auth_algorithm": "sha1",
+			"encapsulation_mode": "tunnel",
+			"encryption_algorithm": "aes-128",
+			"pfs": "group5",
+			"project_id": "b4eedccc6fb74fa8a7ad6b08382b852b",
```
Comment (jtopjian) [inline openstack/networking/v2/extensions/vpnaas/ipsecpolicies/testing/requests_test.go]: Same comment as with the Services List PR. If both `ProjectID` and `TenantID` are being returned, let's make sure both are in the results struct.

**Comment by simonre** [comment]
@jtopjian This is ready for review

**Comment by jtopjian** [comment]
@simonre This looks good! Nice work on the list implementation.

There are some nits in the `doc.go` file. I'm going to merge this and then do a quick commit to fix these + the other one I mentioned, so make sure to pull that commit from master when you continue working on your other PRs.

**Pull request merged**